src/App.tsx

```diff
@@ -1,61 +1,144 @@
 import './App.scss';
+import { useState } from 'react';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const isExist = (todo: Todo) => {
+  return usersFromServer.find(user => user.id === todo.userId) || null;
+};
+
+const getTodos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: isExist(todo),
+}));
+
+const getTodo = (
+  selectedUser: User,
+  title: string,
+  todos: Todo[],
+): Todo => {
+  const lastId = Math.max(...todos.map(todo => todo.id));
+
+  return ({
+    id: lastId + 1,
+    title,
+    completed: false,
+    userId: selectedUser.id,
+    user: selectedUser,
+  });
+};
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(getTodos);
+  const [userError, setUserError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [userSelected, setUserSelected] = useState('0');
+
+  const handleInputTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+
+    setTitleError(false);
+  };
+
+  const handleSelectUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserSelected(event.target.value);
+
+    setUserError(false);
+  };
+
+  const resetForm = () => {
+    setTitle('');
+    setUserSelected('0');
+  };
+
+  const handleAddTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const currentUser = usersFromServer
+      .find(user => user.name === userSelected) || null;
+
+    if (!currentUser) {
+      setUserError(true);
+    }
+
+    if (!title) {
```

Comment:
I can add an empty title by typing just several spaces into the input, I think it shouldn't be like that but if it passes all the tests - it's up to you